src/main/java/org/opensearch/flowframework/transport/WorkflowRequest.java

```diff
@@ -77,6 +105,7 @@ public WorkflowRequest(StreamInput in) throws IOException {
         this.template = templateJson == null ? null : Template.parse(templateJson);
         this.validation = in.readStringArray();
         this.provision = in.readBoolean();
+        this.params = this.provision ? in.readMap(StreamInput::readString, StreamInput::readString) : Collections.emptyMap();
```

Comment:
The `StreamInput` constructor implementation must exactly match the fields written in `writeTo()` where we only write the map in the case `provision` is true:
```java
        if (provision) {
            out.writeMap(params, StreamOutput::writeString, StreamOutput::writeString);
        }
```
We could have used the "optional map" features here but that would have just created a redundant boolean byte in the transport serialization, or just always written the map without that conditional which would have written a variable integer 0 byte for size.  Either way that's an extra byte.

If we're not provisioning, the map is defined to be (and throws an exception if it is not) empty.

Reply:
That makes sense. I was thinking in terms of mutability. Generally, when the map has some entries, using `readMap`can make it as mutable and when it doesn't have anything it can be immutable in nature.

Reply:
> That makes sense. I was thinking in terms of mutability. Generally, when the map has some entries, using `readMap`can make it as mutable and when it doesn't have anything it can be immutable in nature.

You can do `HashMap::new` if you want a mutable map here.